src/zap/zapimage.cpp

```diff
@@ -1588,33 +1588,75 @@ ZapImage::CompileStatus ZapImage::CompileProfileDataWorker(mdToken token, unsign
     return TryCompileMethodDef(token, methodProfilingDataFlags);
 }
 
-void ZapImage::CompileProfileData()
+//  ProfileDisableInlining
+//     Before we start compiling any methods we may need to suppress the inlining
+//     of certain methods based upon our profile data.
+//     This method will arrange to disable this inlining.
+//
+void ZapImage::ProfileDisableInlining()
 {
+    // We suppress the inlining of any Hot methods that have the ExcludeHotMethodCode flag.
+    // We want such methods to be Jitted at runtime rather than compiled in the AOT native image.
+    // The inlining of such a method also need to be suppressed.
+    //
+    ProfileDataSection* methodProfileData = &(m_profileDataSections[MethodProfilingData]);
+    if (methodProfileData->tableSize > 0)
+    {
+        for (DWORD i = 0; i < methodProfileData->tableSize; i++)
+        {
+            CORBBTPROF_TOKEN_INFO * pTokenInfo = &(methodProfileData->pTable[i]);
+            unsigned methodProfilingDataFlags = pTokenInfo->flags;
+
+            // Hot methods can be marked to be excluded from the AOT native image.
+            // We also need to disable inlining of such methods.
+            //
+            if ((methodProfilingDataFlags & (1 << ExcludeHotMethodCode)) != 0)
+            {
+                // Disable the inlining of this method
+                //
+                // @ToDo: Figure out how to disable inlining for this method.               
```

Comment:
Would prefer that this "method is excluded hot" information come through to the jit as a new method attribute, rather than baking in a policy here and (for instance) setting the noinline bit on the method.